flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/FileSystemLoader.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.filesystem;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.core.fs.FileStatus;
+import org.apache.flink.core.fs.FileSystem;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.filesystem.MetaStoreFactory.TableMetaStore;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Optional;
+
+import static org.apache.flink.table.filesystem.FileSystemUtils.generatePartName;
+import static org.apache.flink.table.filesystem.FileSystemUtils.listStatusWithoutHidden;
+
+/**
+ * Loader to temporary files to final output path and meta store. According to overwrite,
+ * the loader will delete the previous data.
+ *
+ * <p>This provide two interface to load:
+ * 1.{@link #loadPartition}: load temporary partitioned files, if it is new partition,
+ * will create partition to meta store.
+ * 2.{@link #loadNonPartition}: just rename all files to final output path.
+ */
+@Internal
+public class FileSystemLoader implements Closeable {
```

Comment:
Rename to FileSystemPartitionLoader or just PartitionLoader?

Reply:
It is not just partition loader, it will load files without partitions.

Reply:
But I am OK to `PartitionLoader`.